mach_eval/analyzers/mechanical/rotor_speed_limit.py

```diff
@@ -0,0 +1,383 @@
+    def __init__(self):
+        pass
+
+    def analyze(self, problem: "SteadyStateStressProblem"):
+        """Analyze method for SPM_RotorSpeedLimitAnalyzer
+    
+        Attributes:
+            problem (SteadyStateStressProblem): SteadyStateStressProblem
+
+        Returns:
+            criterial_stress (np.array): [von_mises_stress,tresca_stress]
+        """
+        
+        sigma_normal = problem.sigma_normal
+
+        # Create Mohrs Circle to compute principle stresses
+        self.mohrs_stress = self.create_mohrs_circle(sigma_normal)
+
+        # Compute Von Mises and Tresca Stress
+
+        criterial_stress = np.array([self.von_mises_stress,self.MSST_stress])
+        return criterial_stress
+
+    @property
+    def von_mises_stress(self):
+        """ Determine Von Mises Equivalent Stress.
+
+        Attributes:
+            mohrs_stress (np.array): Principle Stresses [sigma_1,sigma_2,sigma_3,tau_max]
+
+        Returns:
+            sigma_e (float): Von Mises Equivalent Stress.
+        """
+        sigma_1 = self.mohrs_stress[0]
+        sigma_2 = self.mohrs_stress[1]
+        sigma_3 = self.mohrs_stress[2]
+
+        # Compute Von Mises Equivalent Stress 
+        sigma_e = np.sqrt(2)/2*np.sqrt(
+            (sigma_2-sigma_1)**2+(sigma_3-sigma_1)**2+(sigma_3-sigma_2)**2)
+        
+        return sigma_e
+
+    @property
+    def MSST_stress(self):
+        """ Determine Stress (yield) based on MSST criterion.
+
+        Attributes:
+            mohrs_stress (np.array): Principle Stresses [sigma_1,sigma_2,sigma_3,tau_max]
+
+        Returns:
+            sigma_1-sigma_3 (float): MSST Stress (yield).
+        """
+        # Compute Tresca/MSST stress (yield)
+        sigma_1 = self.mohrs_stress[0]
+        sigma_3 = self.mohrs_stress[2]
+        return sigma_1-sigma_3
+
+    def create_mohrs_circle(self,sigma_normal):
+        """ Determine Principle Stresses (sigma_1, sigma_2, sigma_3) 
+        and Maximum Shear Stress (tau_maximum) using 3D Mohr's Circle Method.
+        ( Applied shear stress terms are omitted in calculation )
+
+        Attributes:
+            sigma_normal (np.array): nominal stress arrays
+        
+        Returns:
+            mohrs_result (np.array): numpy array of mohrs cicrle principle 
+            stresses and maxmimum shear stress.
+        """
+    
+        # Sort arrays for determining principle stresses
+        sigma_normal_s = np.sort(sigma_normal)
+
+        # Split arrays into three arrays
+        sigma_principle = [np.array(x) for x in zip(*sigma_normal_s)] 
+
+        # Obtain Principle Stresses
+        sigma_1 = sigma_principle[2]
+        sigma_2 = sigma_principle[1]
+        sigma_3 = sigma_principle[0]
+
+        # Copmute Maximum Shear Stress
+        tau_max = (sigma_1-sigma_3)/2
+
+        mohrs_stress = np.array([sigma_1,sigma_2,sigma_3,tau_max])
+        return mohrs_stress
+        
+
+class SPM_RotorSpeedLimitResults:
+    def results(
```

Comment:
This should be `__init__()` function. It will have the same inputs as right now, just rename the method